### PR TITLE
updating readme for namespace with nil prefix

### LINKED ISSUE #11: Update README for allowing default XML namespaces and add tests
Update README for allowing default XML namespaces.

Roundtrip test as contributed by @opoudjis:

```ruby
require 'shale'
class Person  < ::Shale::Mapper
  require 'shale/adapter/nokogiri'
  ::Shale.xml_adapter = ::Shale::Adapter::Nokogiri

  attribute :output1, ::Shale::Type::String
  attribute :output2, ::Shale::Type::String
  xml do
    root 'collection'
    namespace 'http://metanorma.org', nil
    map_attribute 'output1', to: :output1
    map_element 'output2', to: :output2
  end
end

person = Person.from_xml(<<~DATA)
  <collection xmlns="http://metanorma.org" output1='A'>
    <output2>John</output2>
    </collection>
DATA

pp person.to_xml
```

_Originally posted by @HassanAkbar in https://github.com/lutaml/shale/issues/6#issuecomment-2134612796_

### PR DESCRIPTION
**Pull request by HassanAkbar**
Updating readme to add documentation related to namespace with nil prefix, and adding test cases for that as well

fixes #11

### CODE REVIEW

**Review by ronaldtse — CHANGES_REQUESTED** [review]
Please use generic namespaces, not Metanorma. Thanks.

**Comment by HassanAkbar** [comment]
@ronaldtse I've updated the code to use generic namespaces

File: Gemfile.lock [hunk 1]
```diff
@@ -13,11 +13,13 @@ GEM
...
+    equivalent-xml (0.6.0)
+      nokogiri (>= 1.4.3)
     json (2.7.1)
     language_server-protocol (3.17.0.3)
-    mini_portile2 (2.8.1)
-    nokogiri (1.13.10)
-      mini_portile2 (~> 2.8.0)
+    nokogiri (1.13.10-arm64-darwin)
+      racc (~> 1.4)
+    nokogiri (1.13.10-x86_64-linux)
```
Comment (ronaldtse) [inline Gemfile.lock]: Have to keep this platform independent.

Comment (HassanAkbar) [reply Gemfile.lock]: @ronaldtse I think there should be no `Gemfile.lock` present in the gem. This file was present in the original gem.
For now I'll revert the changes.

**Pull request merged**